# HG changeset patch
# User Andreas Weber <andy.weber.aw@gmail.com>
# Date 1406023528 -7200
#      Tue Jul 22 12:05:28 2014 +0200
# Node ID 5bfedd39cc77862eff68fa671d46603e008424e3
# Parent  ed3cb9f81145043c000affd9358c922d2ce9d688
Fix indexing in patch if x/y/zdata are row vectors (bug #42822)

* graphics.cc (patch::properties::update_fvc): Get column vectors from
  x/y/zdata if xdata is a row vector.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -7868,17 +7868,19 @@ patch::properties::update_fvc (void)
   dim_vector dv;
   bool is3D = false;
   octave_idx_type nr = xd.rows ();
   octave_idx_type nc = xd.columns ();
   if (nr == 1 && nc > 1)
     {
       nr = nc;
       nc = 1;
-      xd = xd.transpose ();
+      xd = xd.as_column ();
+      yd = yd.as_column ();
+      zd = zd.as_column ();
     }
 
   dv(0) = nr * nc;
   if (zd.is_empty ())
     dv(1) = 2;
   else
     {
       dv(1) = 3;
